File: src/protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/message/type/user/UserTransmit.java
package com.kingstondynamics.dragonsong.protocol.message.type.user;

import com.kingstondynamics.dragonsong.protocol.message.Message;
import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;
import com.kingstondynamics.dragonsong.protocol.utils.Byteinator;
import com.kingstondynamics.dragonsong.protocol.utils.Concatinator;
import com.kingstondynamics.dragonsong.protocol.utils.Keyinator;
import com.kingstondynamics.dragonsong.protocol.utils.Readinator;

public class UserTransmit extends Message {
    
    private final static MessageType type = MessageType.USER_TRANSMIT;
    
    private final String identifier;
    private final String message;
    
    public UserTransmit(String id, String message, String auditID) {
        super(type.getType(), auditID);
        
        this.identifier = id;
        this.message = message;
    }
    
    public UserTransmit(String id, String message) {
        this(id, message, Keyinator.generateGUID());
    }
    
    public UserTransmit(Readinator reader) {
        super(reader);
        
        this.identifier = reader.readIntPrefixedString();
        this.message = reader.readIntPrefixedString();
    }
    
    public UserTransmit(byte[] data) {
        this(new Readinator(data));
    }
    
    public static MessageType getType() {
        return type;
    }
    
    @Override
    public byte[] build() {
        byte[] identifierData = Byteinator.stringToBytesPrefixed(identifier);
        byte[] messageData = Byteinator.stringToBytesPrefixed(message);
        return Concatinator.ConcatinateByteArrays(super.build(), identifierData, messageData);
    }
    
    public String getIdentifier() {
        return identifier;
    }
    
    public String getMessage() {
        return message;
    }
}

File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Messaging/Type/User/UserTransmit.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;
using KingstonDynamics.DragonSong.Protocol.Utils;

namespace KingstonDynamics.DragonSong.Protocol.Messaging.Type.User
{
    public class UserTransmit : Message
    {
        public static readonly MessageType type = MessageType.USER_TRANSMIT;
        
        public readonly string identifier;
        public readonly string message;

        public UserTransmit(string id, string message, string auditId) :  base((int)type, auditId)
        {
            identifier = id;
            this.message = message;
        }

        public UserTransmit(Readinator reader) : base(reader)
        {
            identifier = reader.ReadIntPrefixedString();
            message = reader.ReadIntPrefixedString();
        }

        public UserTransmit(byte[] data) : this(new Readinator(data))
        {
            
        }

        public new byte[] Build()
        {
            var b1 = Byteinator.StringToBytesPrefixed(identifier);
            var b2 = Byteinator.StringToBytesPrefixed(message);
            return Concatinator.ConctatinateByteArrays(base.Build(), b1, b2);
        }
    }
}